common/src/configuration/utils.rs

```diff
@@ -82,7 +82,11 @@ pub fn load_configuration_with_overrides<P: AsRef<Path>, TOverride: ConfigOverri
     };
 
     info!(target: LOG_TARGET, "Configuration file loaded.");
-    let overrides = overrides.get_config_property_overrides(network);
+    let overrides = overrides.get_config_property_overrides(&mut network);
+    // Set the network environment variable according to the chosen network (this is to ensure that
+    // `get_current_or_user_setting_or_default()` uses the environment variable if the static network is not set)
+    std::env::set_var("TARI_NETWORK", network.as_key_str());
```

Comment:
I dont like this.

All this does is going to cause much more confusing, we are changing and setting the network in the configuration loading function?

And to make it worse, `TARI_NETWORK != network` its two different fields

Reply:
Please see improved implementation